libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -754,6 +754,53 @@ void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 		}
 		break;
 	}
+	case FunctionType::Kind::Creation:
+	{
+		solAssert(!functionType->gasSet(), "Gas limit set for contract creation.");
+		solAssert(
+			functionType->returnParameterTypes().size() == 1,
+			"Constructor should return only one type"
+		);
+
+		TypePointers argumentTypes;
+		string constructorParams;
+		for (auto const& arg: arguments)
+		{
+			argumentTypes.push_back(arg->annotation().type);
+			constructorParams += (constructorParams.empty() ? "" : ", ") + IRVariable{*arg}.commaSeparatedList();
+		}
+
+		ContractDefinition const* contract =
+				&dynamic_cast<ContractType const&>(*functionType->returnParameterTypes().front()).contractDefinition();
+
+		Whiskers t(R"X(
```

Comment:
What's `X`?

Reply:
Probably copied from somewhere, I do not really know exact meaning of X here. Removed anyway as not needed.